default/methods/source_control/file/file_browser.py

```diff
@@ -427,6 +427,43 @@ def view_file_list_web_route(project_string_id, username):
                        metadata = file_browser_instance.metadata), 200
 
 
+@routes.route('/api/project/<string:project_string_id>/files', methods = ['GET'])
+@Project_permissions.user_has_project([
```

Comment:
New route, only changes here are the new `query` URL param, and new route path `GET /api/project/<string:project_string_id>/files`